### PR TITLE
fix: update status of Email Campaign

### PR DESCRIPTION
**Pull request by harshkhatri7287**
The status of the email campaign is scheduled to update daily, but it was not updating it in the database,
Now the updated status is directly set in the database also. 
Issue #39179

### CODE REVIEW

File: erpnext/crm/doctype/email_campaign/email_campaign.py [hunk 1]
```diff
@@ -79,10 +79,13 @@ def update_status(self):
 		today_date = getdate(today())
 		if start_date > today_date:
 			self.status = "Scheduled"
+			frappe.db.set_value("Email Campaign", self.name, "status", "Scheduled")
```
Comment (khushrrp) [inline erpnext/crm/doctype/email_campaign/email_campaign.py]: use self.status in place of all status values here.

**Comment by harshkhatri7287** [comment]
@leela @venetanji Please proceed this PR for pending tests.

**Comment by khushali-beehyv** [comment]
@deepeshgarg007 @rohitwaghchaure  Can you please help this PR to proceed for the tests and the review process ?

**Review by barredterra — CHANGES_REQUESTED** [review]
Thanks for the fix! I'm afraid this is not the right place for writing to the database.

Better add `email_campaign.save()` here, after line 146:

https://github.com/frappe/erpnext/blob/22291d32e6dfee445b15a2519e111a781b73abd5/erpnext/crm/doctype/email_campaign/email_campaign.py#L142-L146

**Review by barredterra — APPROVED** [review]

**Comment by harshkhatri7287** [comment]
Hi @barredterra, One required test case is failing, Can you please explain this, why is it happening?

**Pull request merged**